cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/etl/api/TransformContext.java

```diff
@@ -17,11 +17,13 @@
 package co.cask.cdap.etl.api;
 
 import co.cask.cdap.api.annotation.Beta;
+import co.cask.cdap.api.metadata.MetadataReaderContext;
+import co.cask.cdap.api.metadata.MetadataWriterContext;
 
 /**
  * Context passed to ETL Transform stages.
  */
 @Beta
-public interface TransformContext extends StageContext, LookupProvider {
+public interface TransformContext extends StageContext, LookupProvider, MetadataReaderContext, MetadataWriterContext {
```

Comment:
Should this be done for all types of stages? That is, StageContext would extend the two new interfaces. Or how can an action plugin or a error collector manipulate metadata? Do both sources and sinks have a transform context?